### PR TITLE
coastal_boundary dataframe with hourly SCHISM data in Hyfeature unit test 

### PR DESCRIPTION
**Pull request by kumdonoaa**
Previously, multi hourly schism data was used as one of the downstream boundary condition for diffusive routing.  To be consistent with hourly lateral forcing data, hourly SCHISM data has been created from the multi hourly data and preprocess codes for creating coastal boundary depth dataframe has been modified, which includes add downstream boundary file attribute to run_sets.
Note that, for example, total simulation of 4 hrs with max_time_loop of 2hrs requires two sets of lateral forcing in run_sets: the first set of qlat data at t0 and t0+1hr and the second set at t0+2hr and t0+3hr.   However, two sets of SCHISM data needed for this run are, first, SCHISM data at t0, t0+1hr, and t0+2hr and, second, SCHISM data at t0+2hr, t0+3hr, and t0+4hr.

## Additions

-

## Removals

-

## Changes

-

## Testing

1.

## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [ ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: src/troute-network/troute/hyfeature_preprocess.py [hunk 1]
```diff
@@ -741,7 +742,8 @@ def hyfeature_forcing(
     # Assemble coastal coupling data [WIP]
     #---------------------------------------------------------------------
     # Run if coastal_boundary_depth_df has not already been created:
-    if coastal_boundary_depth_df.empty:
+    #if coastal_boundary_depth_df.empty:
```
Comment (shorvath-noaa) [inline src/troute-network/troute/hyfeature_preprocess.py:745]: Should this be un-commented out?

Comment (kumdonoaa) [reply src/troute-network/troute/hyfeature_preprocess.py:745]: It's a temporary solution as for hourly schism data the coastal_boundary_depth_df has to be updated at every time loop, so the lines under the if statement need to be executed at every time loop.  Meanwhile, for multi hourly schism data, the lines need to be executed only once, so the if statement should be there.
